internal/server/models/access_key.go

```diff
@@ -13,20 +13,20 @@ var (
 )
 
 const (
-	ScopePasswordReset        string = "password-reset"
-	ScopeAllowCreateAccessKey string = "create-key"
+	ScopePasswordReset           string = "password-reset"
+	ScopeAllowCreateAccessKey    string = "create-key"
+	AccessKeyForKindUser         string = "user"
+	AccessKeyForKindProvider     string = "provider"
+	AccessKeyForKindOrganization string = "organization"
 )
 
 // AccessKey is a session token presented to the Infra server as proof of authentication
 type AccessKey struct {
 	Model
 	OrganizationMember
-	Name string
-	/* IssuedFor is either:
-	1. The ID of the user that this access key was created for.
-	2. The ID of a provider that is doing SCIM provisioning using this access key.
-	*/
-	IssuedFor     uid.ID
+	Name          string
+	IssuedForID   uid.ID
+	IssuedForKind string
```

Comment:
Is it necessary to use string for kind? It seems the things an access key can be issued for are finite and well known so an enum is likely sufficient

Reply:
I used an int for the `SubjectID `field in grants. It's a `smallint` database type, and in Go it's this: https://github.com/infrahq/infra/blob/af7a6e6cb3ef43fbe3fa22abfe12e33809c68ee5/internal/server/models/grant.go#L39-L50, which is basically a Go version of an enum.

Database enum types could also work, but I think end up being harder to update and change.

Reply:
That's what I had in mind. Not real enums since Go doesn't understand